### DISCUSSION TITLE
Request

### DISCUSSION THREAD

**Discussion by haruo75** [m4mallu/clonebot#72] [General]
Maybe need add Clone Types

copy
forward

**Comment by m4mallu** [comment]
What's the difference between these two ? or, what's the impact on the results when opting these two ?

**Comment by haruo75** [comment]
copy is copy message mean button won't get copied 

forward mean will copy including button caption like how is looks at source channel

**Comment by HithaishiDesai** [comment]
Maybe he meant

Copy - Without Forward Tag
Forward - With Forward Tag


I personally wish Without Forward Tag is best (as you also kept it by default).

Thanks Sir 🙏🏻❤️

**Comment by haruo75** [comment]
yup

**Comment by haruo75** [comment]
![IMG_20220801_222624_409](https://user-images.githubusercontent.com/105282664/182170156-ae2c35eb-d768-48fd-b399-ce4a1ef8bdde.jpg)

this is copy

**Comment by haruo75** [comment]
![IMG_20220801_222622_023](https://user-images.githubusercontent.com/105282664/182170407-43619741-aae7-461e-9696-4068bde1031d.jpg)
this is forward see button also got attached

**Comment by HithaishiDesai** [comment]
But disadvantage of this is if the source Channel/Group is copyrighted all the messages in the target chat stop showing


#solution 

Maybe messages with buttons can be generated via the bot (The one used for cloning) to avoid this above issue [As buttons can only be generated by bot].

Thanks 🙏🏻❤️

**Comment by m4mallu** [comment]
Sir.. In my little knowledge, i know.. a tag will be in files when forwarded from one chat to other :)
But am asking what's the advantage or dis advantage to place this option in clonebot.. ?

**Comment by haruo75** [comment]
https://docs.pyrogram.org/api/methods/copy_message

maybe this will fix forward issues 🤔

**Comment by haruo75** [comment]
> Sir.. In my little knowledge, i know.. a tag will be in files when forwarded from one chat to other :) But am asking what's the advantage or dis advantage to place this option in clonebot.. ?

Advantage
• keep format without change anything

Disadvantage
•Have forward tag
•if main channel file get copyright forward message also get(only if source public channel)
•cant edit message

**Comment by haruo75** [comment]
btw your repo support railway development but you need edit read.me clone words to copy or something,repo name also need change if want to support railway

**Comment by HithaishiDesai** [comment]
> btw your repo support railway development but you need edit read.me clone words to copy or something,repo name also need change if want to support railway

Have deployed it on Railway if you have please share your fork


It's not deploying due to word match of CLONE 😕.

**Comment by haruo75** [comment]
> > btw your repo support railway development but you need edit read.me clone words to copy or something,repo name also need change if want to support railway
> 
> Have deployed it on Railway if you have please share your fork
> 
> It's not deploying due to word match of CLONE 😕.

yup my bot at railway now

**Comment by haruo75** [comment]
fork repo and delete read.me also rename repo

**Comment by HithaishiDesai** [comment]
> > > btw your repo support railway development but you need edit read.me clone words to copy or something,repo name also need change if want to support railway
> > 
> > 
> > Have deployed it on Railway if you have please share your fork
> > It's not deploying due to word match of CLONE 😕.
> 
> yup my bot at railway now

Please share your fork or tell me how to deploy on telegram  @hithaishidesai_605

**Comment by haruo75** [comment]
> > > > btw your repo support railway development but you need edit read.me clone words to copy or something,repo name also need change if want to support railway
> > > 
> > > 
> > > Have deployed it on Railway if you have please share your fork
> > > It's not deploying due to word match of CLONE 😕.
> > 
> > 
> > yup my bot at railway now
> 
> Please share your fork or tell me how to deploy on telegram @hithaishidesai_605

https://github.com/anime-republic/cloner-2

**Comment by HithaishiDesai** [comment]
Thanks Bro 🙏🏻❤️
